external-resolver/src/main/scala/org/genivi/sota/resolver/components/ComponentDirectives.scala

```diff
@@ -21,35 +21,39 @@ import Directives._
 
 /**
  * API routes for creating, deleting, and listing components.
- * @see {@linktourl http://pdxostc.github.io/rvi_sota_server/dev/api.html} 
+ * @see {@linktourl http://pdxostc.github.io/rvi_sota_server/dev/api.html}
  */
 class ComponentDirectives(implicit db: Database, mat: ActorMaterializer, ec: ExecutionContext) {
 
+  def searchComponent =
+    (get & parameter('regex.as[Refined[String, Regex]].?)) { re =>
```

Comment:
the `get`, `post` and `refined` directives need to be assembled outside these methods, because we need to combined them with our authorization directives in ota plus. Some of these can stay here because they will be the same, like `parameter`, or entity, because they are not need to match a path for authentication.